### ISSUE TITLE
重复训练不生成模型的问题

### ISSUE THREAD

**Issue by steamch** [RVC-Boss/GPT-SoVITS#1161]
在我第一次进行微调训练后，会正常生成模型文件。但是当我删除模型之后，不修改设置值，开启SoVITS训练之后只会产生一个模型文件，再点击开启GPT训练，文件夹里连一个模型都没有会直接提示训练完成
`"O:\GPT-SoVITS\runtime\python.exe" GPT_SoVITS/s1_train.py --config_file "O:\GPT-SoVITS\TEMP/tmp_s1.yaml"
Seed set to 1234
Using 16bit Automatic Mixed Precision (AMP)
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
<All keys matched successfully>
ckpt_path: logs\RUDEMAN\logs_s1\ckpt\epoch=14-step=15.ckpt
[rank: 0] Seed set to 1234
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/1
----------------------------------------------------------------------------------------------------
distributed_backend=gloo
All distributed processes registered. Starting with 1 processes
----------------------------------------------------------------------------------------------------

semantic_data_len: 18
phoneme_data_len: 18
                                            item_name                                     semantic_audio
0   aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 271 105 271 105 271 505 396 45 942 735...
1   aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 105 105 105 105 271 505 111 513 402 13...
2   aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 53 105 105 53 271 162 155 306 201 174 ...
3   aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 271 271 271 271 72 368 2 496 624 1013 ...
4   aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 271 536 520 271 72 368 273 753 24 618 ...
5   aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 105 105 105 271 72 505 136 778 204 330...
6   aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 271 105 105 271 72 14 780 781 589 82 3...
7   aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 271 271 271 271 271 505 758 987 846 33...
8   aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 271 271 271 271 72 360 511 466 307 429...
9   aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 105 105 105 271 505 1005 231 808 265 4...
10  aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 271 536 520 271 72 574 588 886 354 971...
11  aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 105 105 105 271 505 490 188 584 998 11...
12  aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 271 271 271 271 72 13 814 320 253 708 ...
13  aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 271 271 105 271 72 405 544 938 51 727 ...
14  aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 105 105 105 271 72 13 107 669 971 971 ...
15  aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 105 105 105 105 271 505 360 173 661 66...
16  aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 271 323 520 271 505 29 879 613 795 188...
17  aaron_mccarlson_scene_aaron_mccarlson_default_...  520 271 105 105 105 105 271 492 441 877 461 97...
dataset.__len__(): 90
O:\GPT-SoVITS\runtime\lib\site-packages\pytorch_lightning\callbacks\model_checkpoint.py:639: Checkpoint directory O:\GPT-SoVITS\logs\RUDEMAN\logs_s1\ckpt exists and is not empty.
Restoring states from the checkpoint path at logs\RUDEMAN\logs_s1\ckpt\epoch=14-step=15.ckpt
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]

  | Name  | Type                 | Params
-----------------------------------------------
0 | model | Text2SemanticDecoder | 77.5 M
-----------------------------------------------
77.5 M    Trainable params
0         Non-trainable params
77.5 M    Total params
309.975   Total estimated model params size (MB)
Restored all states from the checkpoint at logs\RUDEMAN\logs_s1\ckpt\epoch=14-step=15.ckpt
O:\GPT-SoVITS\runtime\lib\site-packages\pytorch_lightning\loops\fit_loop.py:293: The number of training batches (8) is smaller than the logging interval Trainer(log_every_n_steps=50). Set a lower value for log_every_n_steps if you want to see logs for the training epoch.
`Trainer.fit` stopped: `max_epochs=15` reached.`
以上是日志文件。
——————————————————————————————————————
当我修改SoVITS训练的batchsize的值，重新跑一遍会正常生成模型文件
![20240606225439](https://github.com/RVC-Boss/GPT-SoVITS/assets/22424279/0e994d93-1fcf-4651-8d5d-c5236f880519)

**Comment by SapphireLab** [comment]
若只删除了 `GPT_weights` 下的权重但保持相同的实验名，那么依旧会从 `logs/{实验名}/logs_s1` 加载之前的权重文件，所以要重新训练要么换一个实验名，要么删掉 `logs` 文件夹下相应的实验名文件夹。

**Comment by Separatee** [comment]
log看似是记录日志的文件夹，实则不然。在tensorflow等一系列深度学习工具中，中log是用来临时记录模型权重的。如果你只删除了GPT_weight，SoVITS_weight结果输出，是不会让它重新训练的。相反你失去了数据。如果想要效果不好重新训练，请删除log内部所有文件。如果换新语音进行训练，需要删除 **./log/* ** ./input/* ./output内的asr_opt/* slicer_opt/* denoise_opt/*。方可重新训练，

**Comment by steamch** [comment]
@SapphireLab @Separatee 感谢两位大佬的讲解，问题已经解决

**Comment by gffffsds** [comment]
谢谢问题解决
